src/parser.cpp

```diff
@@ -11,6 +11,16 @@ namespace vpd
 Parser::Parser(const std::string& vpdFilePath, nlohmann::json parsedJson) :
     m_vpdFilePath(vpdFilePath), m_parsedJson(parsedJson)
 {
+    if (!std::filesystem::exists(m_vpdFilePath) ||
+        std::filesystem::file_size(m_vpdFilePath) == 0)
```

Comment:
Handle exception for these calls as well.

Reply:
Done